### PR TITLE
Fixes message generation to work with the new rosjava_bootstrap.

### PR DESCRIPTION
**Pull request by damonkohler**
Also sort and remove duplicate package.xml dependencies.

### CODE REVIEW

**Comment by SirVer** [comment]
lgtm.

**Comment by stonier** [comment]
This rather breaks our release builds.

File: build.gradle [hunk 1]
```diff
@@ -15,51 +15,67 @@
...
 task wrapper(type: Wrapper) {
-  gradleVersion = '1.8'
+  gradleVersion = "1.9"
 }
 
 buildscript {
-    def rosMavenPath = "$System.env.ROS_MAVEN_PATH".split(':').collect { 'file://' + it }
-    def rosMavenRepository = "$System.env.ROS_MAVEN_REPOSITORY"
-    if ( rosMavenRepository == 'null' ) {
-        rosMavenRepository = "https://github.com/rosjava/rosjava_mvn_repo/raw/master"
```
Comment (stonier) [inline build.gradle:25]: If you lose this, you lose the ability to just call gradle on it as per the rosjava_core docs: see https://github.com/rosjava/rosjava_core/issues/186

File: build.gradle [hunk 2]
```diff
@@ -15,51 +15,67 @@
...
+    } else {
+      maven {
+        url "https://github.com/rosjava/rosjava_mvn_repo/raw/master"
+      }
+    }
+    mavenCentral()
+  }
+
+  dependencies {
+    classpath group: "org.ros.rosjava_bootstrap", name: "gradle_plugins", version: "[0.2,0.3)"
```
Comment (stonier) [inline build.gradle:53]: Hydro builds will fail with this. It looks like we need a master branch after all which should get occasionally synced with hydro branch dependency updates.

File: build.gradle.template [hunk 3]
```diff
@@ -14,19 +14,17 @@
  * the License.
  */
 
-try {
-    project.catkin.tree.generateMessageArtifact(project, '@name@')
-} catch (NullPointerException e) {
-    println("Couldn't find @name@ on the ROS_PACKAGE_PATH")
+if (project.catkin.tree.pkgs.containsKey("@name@")) {
+  project.catkin.tree.pkgs.@name@.generateMessageArtifact(project)
```
Comment (stonier) [inline build.gradle.template:18]: This is reverting to an old api which will deprecate.

File: build.gradle (hunk 1 above)
Comment (damonkohler) [inline build.gradle:25]: This is actually fixed by adding the appropriate spring source repos.

File: build.gradle (hunk 2 above)
Comment (damonkohler) [inline build.gradle:53]: Agreed. Done.

File: build.gradle.template (hunk 3 above)
Comment (damonkohler) [inline build.gradle.template:18]: I'll nuke the deprecated stuff and move this method out of CatkinPackage to be more like your PR13

**Comment by damonkohler** [comment]
Sorry for the build breakage. I've rolled back and moved on to developing in a new master branch.

**Pull request merged**